clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -5333,11 +5333,31 @@ bool Sema::CheckIfFunctionSpecializationIsImmediate(FunctionDecl *FD,
   return false;
 }
 
+static QualType GetImplicitObjectParameterTypeCXX20(ASTContext &Context,
```

Comment:
I don't think splitting this method up into 2 like this is necessary, I think these two should be the same function with the language-based split for the condition of the 'if' statement.